fs/sync/sync.go

```diff
@@ -557,20 +560,44 @@ func (s *syncCopyMove) srcParentDirCheck(entry fs.DirEntry) {
 	}
 }
 
-// renameHash makes a string with the size and the hash for rename detection
+// isUsingRenameStrategy checks if a strategy is included in a list of strategies
+func isUsingRenameStrategy(strategy string, strategies []string) bool {
+	for _, s := range strategies {
+		if s == strategy {
+			return true
+		}
+	}
+	return false
+}
+
+// renameID makes a string with the size and the other identifiers of the requested rename strategies
 //
 // it may return an empty string in which case no hash could be made
-func (s *syncCopyMove) renameHash(obj fs.Object) (hash string) {
-	var err error
-	hash, err = obj.Hash(s.ctx, s.commonHash)
-	if err != nil {
-		fs.Debugf(obj, "Hash failed: %v", err)
-		return ""
+func (s *syncCopyMove) renameID(obj fs.Object, renameStrategy []string, precision time.Duration) (renameID string) {
+	renameID = ""
+	renameID += fmt.Sprintf("%d", obj.Size())
+
+	if isUsingRenameStrategy("hash", renameStrategy) {
+		var err error
+		hash, err := obj.Hash(s.ctx, s.commonHash)
+
+		if err != nil {
+			fs.Debugf(obj, "Hash failed: %v", err)
+			return ""
+		}
+		if hash == "" {
+			return ""
+		}
+
+		renameID += fmt.Sprintf(",%s", hash)
 	}
-	if hash == "" {
-		return ""
+
+	if isUsingRenameStrategy("modtime", renameStrategy) {
+		modTime := obj.ModTime(s.ctx).Unix() / precision.Nanoseconds() / 2
```

Comment:
If `precision.Nanoseconds()` is small (eg it is 1 on local unix drives) then it would be nice to include the nanoseconds too in the comparison.

So the calculation really wants to be something like this (with rounding)

```go
modTime := obj.ModTime(s.ctx)
divisor := precision.Nanoseconds() / 2
rounding := divisor / 2
if rounding > 0 {
    rounding--
}
timeHash := (modTime.Unix() * time.Nanosecond + modTime.Nanosecond() + rounding ) / divisor
```

Note that `modTime.Unix() * time.Nanosecond` potentially overflows an int64. It will do so when modTime.Unix > 2**63/1E9 = the year 2262 so that is probably OK... If it isn't we'll have to break out the large integer maths library.

Note also we should probably be rejecting "modtime" if the `Precision == fs.ModTimeNotSupported`

Reply:
Okay, I have pushed another commit which resolves the issues you have pointed out. For the last suggestion, rejecting the modtime, I have handled it as it was previously handled when hashes are not available in either remote. So if one remote doesn't support modtime, (the window is ModTimeNotSupported) then `s.trackRenames` will be set to false.